src/TimePicker/TimePicker.tsx

```diff
@@ -131,41 +138,72 @@ export const createTimeList = ({ hours, minutes }, timeFormat) => {
   }
 };
 
-const startsWith = (time, originalInput) => {
+const startsWith = (time: TimeAndFormat, originalInput: string) => {
   const timeStr = formatTime(time);
   return timeStr.substr(0, originalInput.length) === originalInput;
 };
 
-export const filterTime = ({ originalInput, minTime, maxTime }) => time => {
+type FilterTimeInput = {
+  originalInput: string
+  minTime: Time
+  maxTime: Time
+};
+export const filterTime = ({ originalInput, minTime, maxTime }: FilterTimeInput) => (time: TimeAndFormat) => {
   return lteTime(minTime, time) && lteTime(time, maxTime) && startsWith(time, originalInput);
 };
 
-export const makeOptions = ({ minTime, maxTime, timeFormat, userValue }, inputValue) => {
+type MakeOptionsInput = {
+  timeFormat: TimeFormat
+  minTime: Time
+  maxTime: Time
+  userValue?: Time
+};
+export const makeOptions = ({ minTime, maxTime, timeFormat, userValue }: MakeOptionsInput, inputValue: string) => {
   const time = parseInTimeFormat(inputValue, timeFormat);
   const selectedValue = (userValue && userValue !== inputError) ? { ...userValue, timeFormat } : userValue;
-  const timeList = (time === inputError ? [] : createTimeList(time, timeFormat)).concat(compact([selectedValue]));
+  const timeList = (
+    (time === inputError ? [] : createTimeList(time as Time, timeFormat)) as (Time | TimeAndFormat)[] // TODO(typo)
+  ).concat(compact([selectedValue]));
   const filteredTimeList = timeList.filter(filterTime({
-    originalInput: time.originalInput || '', minTime, maxTime
+    originalInput: (time as any).originalInput || '', minTime, maxTime // TODO(typo)
   }));
   return uniqBy(sortBy(filteredTimeList.map(toOption), 'value'), 'value');
 };
 
+export type OnChangeInput = {
+  originalInput?: string
+} & Partial<Time>;
+
+export interface RequiredProps {
```

Comment:
the naming is different than the conventions we're using other components, see #1082 

Also we've decided to export types, not interfaces.